src/librbd/io/CopyupRequest.cc

```diff
@@ -291,7 +291,7 @@ void CopyupRequest<I>::update_object_maps() {
   auto cct = m_image_ctx->cct;
   ldout(cct, 20) << "oid=" << m_oid << dendl;
 
-  if (!m_image_ctx->snaps.empty()) {
+  if (!m_image_ctx->snaps.empty() && !m_copyup_is_zero) {
```

Comment:
... again, the deep-copy is the odd case where we *might* need to update the object map snapshots if deep-copy actually copied data (i.e. the object exists).